### PR TITLE
SWAN GPU

### PR DESCRIPTION
**Pull request by omazapa**
Hi Guys,

this is the code to install the nvidia-docker2 middleware 
and to check if the nvidia is loaded.
The driver should be installed in the host by the user.

Cheers
Omar.

### CODE REVIEW

File: SetupInstall-Centos7.sh [hunk 1]
```diff
@@ -4,6 +4,22 @@
...
 
 
+# ----- Install the required gpu software on the host ----- #
+install_gpu_software()
+{
+  echo "Installing nvidia-docker2..."
+  # Add the package repositories
+  distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
+  curl -s -L https://nvidia.github.io/nvidia-docker/$distribution/nvidia-docker.repo | \
+  sudo tee /etc/yum.repos.d/nvidia-docker.repo
```
Comment (ebocchi) [inline SetupInstall-Centos7.sh]: Do we really need to add the repo file?
Cannot we just install the package without messing around with the local manager config?

File: SetupInstall-Centos7.sh [hunk 2]
```diff
@@ -30,6 +46,21 @@ install_software()
...
+
+echo ""
+read -r -p "Do you want to proceed with the gpu software installation [y/N] " response
+case "$response" in
+  [yY]) 
+    echo "Installing required gpu software..."
+    install_gpu_software
+  ;;
+  *)
+    echo "Exiting..."
```
Comment (ebocchi) [inline SetupInstall-Centos7.sh:64]: If I say no, I can still use the whole machinery without GPUs.
There is no need to bail out. And, in fact, you are not bailing out...
Please, echo something like "Continuing without GPU support"

File: SetupInstall-Ubuntu.sh [hunk 3]
```diff
@@ -3,6 +3,26 @@
...
+install_gpu_software()
+{
+
+  echo "Installing nvidia-docker2..."
+  # Add the package repositories
+  curl -s -L https://nvidia.github.io/nvidia-docker/gpgkey | \
+  sudo apt-key add -
+  distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
+  curl -s -L https://nvidia.github.io/nvidia-docker/$distribution/nvidia-docker.list | \
+  sudo tee /etc/apt/sources.list.d/nvidia-docker.list
```
Comment (ebocchi) [inline SetupInstall-Ubuntu.sh]: Same comment as for CC7. Cannot we simply install the package?

**Review by ebocchi — CHANGES_REQUESTED** [review]

**Comment by ebocchi** [comment]
Thanks a lot, @omazapa !
I have a couple of minor comments on the code.
Is there any upstream-provided list of the compatible hardware?
It would be good to tell people to check if their machine is supported beforehand.

Cheers!

**Comment by omazapa** [comment]
Please don't do the review yet,
I am testing it in a clean installation of Centos7 and ubuntu.
I will be uploading the changes.

**Comment by omazapa** [comment]
Hi @ebocchi 
I did the changes and It was  tested it in Centos 7 and Ubuntu 16.04/18.04 
You can do the review again.

Thank you for the review!

**Comment by omazapa** [comment]
Hi @ebocchi 
About the hardware https://docs.nvidia.com/deploy/cuda-compatibility/index.html#support-hardware
I can see that all architectures are supported except the video card with Fermi.

Cheers.
Omar.


> Thanks a lot, @omazapa !
> I have a couple of minor comments on the code.
> Is there any upstream-provided list of the compatible hardware?
> It would be good to tell people to check if their machine is supported beforehand.
> 
> Cheers!

About the hardware, appear to be compatible with any card compatible with cuda 10

File: SetupInstall-Centos7.sh (hunk 1 above)
Comment (ebocchi) [reply SetupInstall-Centos7.sh]: Fixed.

File: SetupInstall-Ubuntu.sh (hunk 3 above)
Comment (ebocchi) [reply SetupInstall-Ubuntu.sh]: Fixed.

**Pull request merged**